### PR TITLE
Migrate from Jenkins to GitHub Actions

### PR DESCRIPTION
**Pull request by negz**
See https://github.com/crossplane/crossplane/pull/1865 for context. Note that I've also attempted to merge the existing GitHub e2e workflow into the standard one we're using for all Crossplane projects, since the existing one was doing duplicate work.

If these changes look good, I will:

* Remove the webhook that triggers the Jenkins build
* Mark the Jenkins and 'Build' checks as not required
* Mark the lint, check-diff, and unit-tests checks as required

I notice the e2e tests seem flaky. Should I also mark them required?

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,252 @@
...
+      - name: Vendor Dependencies
+        run: make vendor vendor.check
+
+      - name: Build Helm Chart
+        run: make -j2 build
+        env:
+          # We're using docker buildx, which doesn't actually load the images it
+          # builds by default. Specifying --load does so.
+          BUILD_ARGS: "--load"
+
```
Comment (hasheddan) [inline .github/workflows/ci.yml:169]: Looks like just adding https://github.com/crossplane/oam-kubernetes-runtime/blob/723f8ecf8444e0feed1493156626df6d54d0c116/.github/workflows/go.yml#L46 may do the trick

**Comment by wonderflow** [comment]
This PR is really cool! @negz Thanks!

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,279 @@
...
+      - name: Vendor Dependencies
+        run: make vendor vendor.check
+
+      - name: Check Diff
+        run: make check-diff
+
+  unit-tests:
+    runs-on: ubuntu-18.04
+    needs: detect-noop
+    if: needs.detect-noop.outputs.noop != 'true'
```
Comment (wonderflow) [inline .github/workflows/ci.yml:107]: can you explain a bit what will it help? I see lots of this lines, but not very understand why we need it.  thanks

**Review by wonderflow — APPROVED** [review]
Thanks @negz , this is really cool Job. 

e2e test is also required, we don't know why it's sometimes broken, we can't re-produce it locally, I assume it's because of lack of resource?

File: .github/workflows/ci.yml (hunk 2 above)
Comment (negz) [reply .github/workflows/ci.yml:107]: This is just a little optimisation that skips running CI (i.e. automatically passes CI) for PRs that either:

* Only touch markdown files or images.
* Are identical to another PR that already passed CI (i.e. the content of the two refs would have matching hashes).

**Pull request merged**